question-service/src/routes/deleteQuestionsController.ts

```diff
@@ -0,0 +1,38 @@
+import express, { Request, Response } from 'express';
+import { Collection } from 'mongodb';
+import { connectToDB } from '../db/mongoClient';
+
+const router = express.Router();
+let questionsCollection: Collection;
+
+// Middleware to connect to MongoDB and get the collection
+router.use(async (_, res, next) => {
+  try {
+    const db = await connectToDB();
+    questionsCollection = db.collection('questions');
+    next();
+  } catch (err) {
+    console.error('MongoDB connection error:', err);
+    res.status(500).json({ error: `Failed to connect to MongoDB: ${err}` });
+  }
+});
+
+// DELETE a question
+router.delete('/:questionId', async (req: Request, res: Response) => {
+  const { questionId } = req.params;
```

Comment:
The routes will be confusing as we do not know whether the delete is for user or for userquestion